### ISSUE TITLE
ofToHex(string) is wrong for non-ascii input

### ISSUE THREAD

**Issue by pappis** [openframeworks/openFrameworks#1173]
If you pass a string containing non-ascii characters to ofToHex(), the non-ascii characters (with character codes above 127) will be output as ffffxxxx (it depends on the exact character value and the size of the int type on the platform you are compiling for).
The problem is due to an incorrect cast in the ostringstream operation inside the loop.
Instead of directly casting the (signed) char from the input character to an unsigned int, the input character must first be cast to an unsigned char.
In other words, instead of:
`out << setfill('0') << setw(2) << hex << (unsigned int) value[i];`
it should be:
`out << setfill('0') << setw(2) << hex << (unsigned int) ((unsigned char)value[i]);`
or maybe even:
`out << setfill('0') << setw(2) << hex << static_cast<unsigned int>(static_cast<unsigned char>(value[i]));`
if you prefer C++ style casts over C style casts. :)

HTH
/Jesper
